torch/autograd/graph.py

```diff
@@ -411,6 +411,82 @@ def __setstate__(self, state):
     return Handle(tuple(handles))
 
 
+def _register_multi_post_grad_hook(tensors: Sequence[torch.Tensor], fn: Callable[[Sequence[Optional[torch.Tensor]]], None]):
+    r"""Registers a multi-post-grad backward hook.
+
+    This is similar to :meth:`register_multi_grad_hook` except the hook runs
+    after each tensor's gradient has been accumulated, not just computed.
+    """
+    count: Dict[int, int] = dict()
+    nb_calls = None
+    buffer: Dict[int, List[Optional[torch.Tensor]]] = dict()
+
+    def get_grad_fn(t):
+        # or grad accumulator
+        if t.requires_grad and t.grad_fn is None:
+            return t.expand_as(t).grad_fn.next_functions[0][0]
+        else:
+            return t.grad_fn
+
+    grad_fns = list(map(get_grad_fn, tensors))
```

Comment:
is it better to use a comprehension directly? `[get_grad_fn(t) for t in tensors]`?